toolkit/tools/pkg/imagecustomizerlib/customizeuki.go

```diff
@@ -624,3 +730,140 @@ func getKernelNameFromUki(ukiPath string) (string, error) {
 	kernelName := "vmlinuz-" + matches[1]
 	return kernelName, nil
 }
+
+func extractSectionFromUkiWithObjcopy(ukiPath string, sectionName string, outputPath string, buildDir string) error {
+	tempCopy, err := os.CreateTemp(buildDir, "uki-copy-*.efi")
+	if err != nil {
+		return fmt.Errorf("failed to create temp UKI copy:\n%w", err)
+	}
+	defer os.Remove(tempCopy.Name())
+	tempCopy.Close()
+
+	input, err := os.ReadFile(ukiPath)
+	if err != nil {
+		return fmt.Errorf("failed to read UKI file:\n%w", err)
+	}
+	if err := os.WriteFile(tempCopy.Name(), input, 0o644); err != nil {
```

Comment:
Use `tempCopy.Write()` here instead. (You will need to move the `tempCopy.Close()` call.)